tests/Aspire.Hosting.Azure.Tests/AzureResourcePreparerTests.cs

```diff
@@ -235,6 +235,38 @@ public async Task NullCommandLineArgIsIgnored()
         Assert.True(true);
     }
 
+    [Fact]
+    public async Task CommandLineArgsCallbackContextHasCorrectExecutionContextDuringPublish()
+    {
+        using var builder = TestDistributedApplicationBuilder.Create(DistributedApplicationOperation.Publish);
+        builder.AddAzureContainerAppEnvironment("env");
+
+        // Create a project with a WithArgs callback that checks IsRunMode before accessing ServiceProvider
+        // This simulates what WithVSCodeDebugSupport does
+        var api = builder.AddProject<Project>("api", launchProfileName: null)
+            .WithArgs(context =>
+            {
+                // This simulates WithVSCodeDebugSupport behavior - checking IsRunMode before accessing ServiceProvider
+                if (!context.ExecutionContext.IsRunMode)
+                {
+                    return;
+                }
+
+                // This should not be reached during publish mode, but if the ExecutionContext is not set correctly,
+                // it will default to Run mode and then try to access ServiceProvider which will throw
+                _ = context.ExecutionContext.ServiceProvider;
```

Comment:
1. we don't need the IsRunMode check. That can be removed.
2. You can set the `IServiceProvider` to a variable outside of the callback on line 243, and then Assert that it is not null at the end of the test on line 267.